test/service/src/main.rs

```diff
@@ -0,0 +1,192 @@
+// Copyright 2019-2021 Parity Technologies (UK) Ltd.
+// This file is part of Cumulus.
+
+// Cumulus is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Cumulus is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Cumulus.  If not, see <http://www.gnu.org/licenses/>.
+mod cli;
+
+use clap::Parser;
+use cli::{Commands, RelayChainCli, TestCollatorCli};
+use cumulus_client_service::genesis::generate_genesis_block;
+use cumulus_primitives_core::{relay_chain::v2::CollatorPair, ParaId};
+use cumulus_test_service::AnnounceBlockFn;
+use polkadot_service::runtime_traits::AccountIdConversion;
+use sc_cli::{CliConfiguration, SubstrateCli};
+use sp_core::{hexdisplay::HexDisplay, Encode, Pair};
+use sp_runtime::traits::Block;
+use std::{io::Write, sync::Arc};
+
+fn extract_genesis_wasm(chain_spec: &Box<dyn sc_service::ChainSpec>) -> Result<Vec<u8>, String> {
+	let mut storage = chain_spec.build_storage()?;
+
+	storage
+		.top
+		.remove(sp_core::storage::well_known_keys::CODE)
+		.ok_or_else(|| "Could not find wasm file in genesis state!".into())
+}
+
+pub fn wrap_announce_block() -> Box<dyn FnOnce(AnnounceBlockFn) -> AnnounceBlockFn> {
+	Box::new(|_| {
+		// Never announce any block
+		Arc::new(|_, _| {})
+	})
+}
```

Comment:
This is not used anywhere; I'm assuming you'll add a command line argument for this, right? (For the `pov_recovery` test.)